### ISSUE TITLE
Outdated Detours, x64, and error code 0xC0000142

### ISSUE THREAD

**Issue by jonbarrow** [tarekwiz/CurlHook#1]
The Detours build used here, 1.5, is very outdated. The latest is 4.0.1

There also seems to be no way of building this project for x64, as changing the platform to x64 fails to build due to the included detours lib being x86

I have ripped the relevant parts of this project that I need (to update to the latest Detours and build for x64) which builds properly, however the process crashes. In my case the cURL functions are never exported in the program I need to hook, so the "pattern scanning" method is required, however the process crashes with error code `0xC0000142`. This happens inside `FindPattern`, on the line `found &= mask[j] == '?' || pattern[j] == *(char*)(base + i + j);`, and it is specifically caused by `*(char*)(base + i + j)` (as commenting out `|| pattern[j] == *(char*)(base + i + j)` prevents the crash, but obviously does not fix anything)

In my case I am trying to disable SSL verification in the Cemu emulator (http://cemu.info/) in order to redirect some https requests to my own custom servers. The redirection is working, but now SSL verification is in the way

**Comment by tarekwiz** [comment]
1. Detour version has nothing to do with how the program would perform, 1.5 is perfectly fine.
2. You literally commented out the pattern scanning function itself. you need to open IDA, or any reverse engineering program and find the function you're trying to hook, then make a pattern for it. The pattern is probably what's failing. I suspect that the curl functions that you're trying to hook are not even exported, so you need to find out which of these patterns break and maybe find a better pattern for it

**Comment by jonbarrow** [comment]
The mention of the detours version was more so just wondering why you went with such an outdated version, not an actual issue with the project

> You literally commented out the pattern scanning function itself

Yes, I am ware. However that was not the issue. When I said I commented part of it out, it was to bring attention to the fact that the code that piece of code was the part causing issues, not that commenting it out was the cause

However I have fixed my issues using my own methods, so I will close the now. Thank you for your time, and nice project
